pkg/security/security_profile/dump/activity_dump.go

```diff
@@ -279,6 +279,37 @@ func NewActivityDumpFromMessage(msg *api.ActivityDumpMessage) (*ActivityDump, er
 	return ad, nil
 }
 
+// computeSyscallsList returns the aggregated list of all syscalls
+func (ad *ActivityDump) computeSyscallsList() []uint32 {
+	mask := make(map[int]uint32)
+	nodes := ad.ProcessActivityTree
+	var node *ProcessActivityNode
+	if len(nodes) > 0 {
+		node = nodes[0]
+	}
+
+	for node != nil {
+		for _, nr := range node.Syscalls {
+			mask[nr] = 1
+		}
+		for _, child := range node.Children {
+			nodes = append(nodes, child)
+		}
+		nodes = append(nodes[1:])
```

Comment:
Not sure, but this seems wrong

Reply:
I think it does what's it's supposed to ... I can move things around but it will eventually do the same thing. Adding unit tests !

Reply:
I mean, what does `nodes = append(nodes[1:])` mean ? Isn't the same as `nodes = nodes[1:]`?

Reply:
I changed the algo a little bit to make it clearer 😃